### DISCUSSION TITLE
More guidance for recursive types

### DISCUSSION THREAD

**Discussion by carueda** [scala-tsi/scala-tsi#214] [Q&A] [answered]
First off, thanks for this tool, seems very helpful!

However, I'm having trouble trying to determine what to do so scala-tsi can handle a Json AST case like spray-json's [JsValue](https://github.com/spray/spray-json/blob/release/1.3.x/src/main/scala/spray/json/JsValue.scala#L27).

As an example:

```scala
package somepkg

import spray.json.JsValue

case class FooWithJsValue(
    foo:     String,
    jsValue: JsValue,
)
```

Unsurprisingly, as indicated in the README regarding [lack of support for circular references](https://github.com/scala-tsi/scala-tsi#circular-references), the plugin misbehaves (that is, hanging/not making any progress) eventually triggering java.lang.OutOfMemoryError.

In particular, I'm not sure how to put together the suggested implicit-based values to "help" scala-tsi make sense of this (3rd party) sealed type.  If actually possible with scala-tsi, could anyone provide some additional guidance to handle cases like this?  Thanks in advance.

**Comment by dhoepelman** [comment, marked as answer]
You need to make an implicit TypescriptType and import it through SBT. See `MyModelTSTypes`  in the README
Feel free to open a PR to clarify the documentation if you think it could be improved.

[Here is a JSON type you can use](https://github.com/scala-tsi/scala-tsi/pull/215/files#diff-1fa089a7e7682905e97e916783d5c10644be7892bad0877f1423e4f0ba5f1986), so you can define a `implicit val jsValueTsType: TSType[spray.json.JsValue] = TSType(JsonValue)` and import it

**Reply by carueda** [reply]
👍🏼  Great, I'm now able to handle JsValue as well as other types that end up being serialized as JSON. In short:

```scala
object ForTypescript {
  implicit val JsValueTsType: TSType[JsValue] = TSType(CommonTypes.JsonValue)
  implicit val FeatureTsType: TSType[Feature] = TSType(CommonTypes.JsonValue)
  implicit val GeometryTsType: TSType[Geometry] = TSType(CommonTypes.JsonValue)
}
```
Many thanks for the extra hints!
